hazelcast/src/test/java/com/hazelcast/NoMixedJUnitAnnotationsInOurTestSourcesTest.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2008-2023, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast;
+
+import com.hazelcast.test.archunit.ArchUnitRules;
+import com.hazelcast.test.archunit.ArchUnitTestSupport;
+import com.tngtech.archunit.core.domain.JavaClasses;
+import com.tngtech.archunit.core.importer.ClassFileImporter;
+import org.junit.Test;
+
+import static com.hazelcast.test.archunit.ModuleImportOptions.onlyCurrentModuleTests;
+
+public class NoMixedJUnitAnnotationsInOurTestSourcesTest extends ArchUnitTestSupport {
```

Comment:
What about other maven modules?

Reply:
The other modules for example hazelcast-sql, hazelcast-tpc-engine 
do not have dependency for archunit in their pom.xml files. None of them have archunit rule. I am not sure why

Reply:
rev : 7dbaaa4de9a8696be18007b93ebbc488fb133d3e